ledger/src/blockstore.rs

```diff
@@ -2213,6 +2291,30 @@ impl Blockstore {
         })
     }
 
+    fn get_any_valid_slot_entries(&self, slot: Slot, start_index: u64) -> Vec<Entry> {
+        let (completed_ranges, slot_meta) = self
+            .get_completed_ranges(slot, start_index)
+            .unwrap_or_default();
+        if completed_ranges.is_empty() {
+            return vec![];
+        }
+        let slot_meta = slot_meta.unwrap();
+
+        let entries: Vec<Vec<Entry>> = PAR_THREAD_POOL_ALL_CPUS.with(|thread_pool| {
```

Comment:
Maybe we can extract out this `par_iter` logic into a function that takes a thread pool argument, so it can be reused within `get_slot_entries_with_shred_info`

Reply:
This turned out to be a bit of a pain, since we return a `Result` in one case and just the `Vec` in another. Okay to leave as-is?

Reply:
Yeah no worries